sql/src/main/java/io/crate/analyze/repositories/RepositorySettingsModule.java

```diff
@@ -71,22 +70,19 @@
             .build()) {
 
         @Override
-        public Optional<GenericProperties> dynamicProperties(Optional<GenericProperties> genericProperties) {
-            if (!genericProperties.isPresent()) {
+        public GenericProperties dynamicProperties(GenericProperties genericProperties) {
+            if (genericProperties.isEmpty()) {
                 return genericProperties;
             }
-            GenericProperties dynamicProperties = null;
-            Map<String, Expression> properties = genericProperties.get().properties();
+            GenericProperties dynamicProperties = new GenericProperties();
+            Map<String, Expression> properties = genericProperties.properties();
```

Comment:
You could also avoid the declaration of the variable here and iterate on the entrySet directly.